_sc2data/Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ValidatorData.xml

```diff
@@ -320,6 +342,409 @@
         <CombineArray value="AP_IsVanillaCommandCenterFlying"/>
         <CombineArray value="AP_IsAPCommandCenterFlying"/>
     </CValidatorCombine>
+    <CValidatorCombine id="AP_IsInvulnerableAndHasSOAHeroicShieldImmunity">
+        <Type value="And"/>
+        <CombineArray value="AP_IsInvulnerable"/>
+        <CombineArray value="AP_HasSOAHeroicShieldImmunity"/>
+    </CValidatorCombine>
+    <CValidatorUnitType id="AP_IsSOACaster">
+        <Value value="AP_SoACaster"/>
+    </CValidatorUnitType>
+    <CValidatorUnitType id="AP_IsVanillaSOACaster">
+        <Value value="SoACaster"/>
+    </CValidatorUnitType>

```

Comment:
This should belong to `ArchipelagoPlayerLotV` instead as `SoACaster` doesn't exist here

Reply:
I have yet to implement AP_SOAcaster, I'm leaving that for later. Doesn't hurt it to have a validator here already though?

Reply:
This validator doesn't belong to this mod, it should be part of `ArchipelagoPlayerLotV`

Reply:
I'm not sure why...?  Can't AP_SOA be a part of `ArchipelagoPlayer`?